sources/packages/backend/apps/api/src/route-controllers/education-program/education-program.controller.service.ts

```diff
@@ -98,7 +99,15 @@ export class EducationProgramControllerService {
       await this.institutionService.getInstitutionTypeById(institutionId);
     payload.isBCPrivate = institutionType.isBCPrivate;
     payload.isBCPublic = institutionType.isBCPublic;
-
+    if (programId) {
+      const educationProgram =
+        await this.educationProgramService.getActiveEducationProgram(programId);
+      if (!educationProgram) {
+        throw new UnprocessableEntityException(
+          "Education Program is not active. Not able to a save the program due to an invalid request.",
+        );
+      }
+    }
```

Comment:
Unless I missing something, validating program on save is already existing and not a requirement for this ticket.

![image](https://github.com/bcgov/SIMS/assets/54600590/ac9ba62b-8298-4873-a321-1e950a5bfa37)